compiler/hash-driver/src/driver.rs

```diff
@@ -0,0 +1,358 @@
+use std::{collections::HashMap, env::current_dir, ops::Deref, path::Path, time::Duration};
+
+use hash_ast::node_map::{InteractiveBlock, ModuleEntry};
+use hash_pipeline::{
+    fs::{read_in_path, resolve_path, PRELUDE},
+    interface::{CompilerInterface, CompilerResult, CompilerStage},
+    settings::CompilerStageKind,
+};
+use hash_reporting::writer::ReportWriter;
+use hash_source::{ModuleKind, SourceId};
+use hash_utils::{log, stream_writeln, timing::timed};
+
+/// The Hash Compiler interface. This interface allows a caller to create a
+/// [Compiler] with the specified components. This allows external tinkerers
+/// to add their own implementations of each compiler stage with relative ease
+/// instead of having to scratch their heads.
```

Comment:
not clear how to do this yet?

Reply:
Yeah, this is an old comment, which should eventually be realised, should we keep this or no?

Reply:
Maube add a `@@Future` to it?